### DISCUSSION TITLE
EVOHD+ connection issues

### DISCUSSION THREAD

**Discussion by dsge80** [ParadoxAlarmInterface/pai#409] [Q&A] [answered]
Hello. Since one week,  from when I've bought the alarm system, I'm trying to integrate it in HA with PAI, using IP150+.
I've bought it with PAI in mind, but it was advertised as EVOHD (non plus version) and is the + version.

Anyway I've try it different PAI versions with almost same result, password is ok for sure as I'm using it also in Babyware and doesn't contains "0".

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/fe1b617d-9eaa-4f3f-bf43-2b77c7107361)

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/3e504ce3-dc87-49e4-9e58-50cc6355ab5e)

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/f546bc4e-ad17-44fa-a914-2a26b3e71dc8)

I know is still in development stage, so I don't know if I'm doing something wrong or simply is a communication problem. 
By example on serial1 I do have some issues to se IP150 itself in Babyware but are not any errors  on serial2.
I'm beginner on alarms systems and also on PAI so I'm trying to fix it in  the next days, how long can replace the board, in case would be necessary.
Any help is  appreciated. 

Thank you.

**Comment by yozik04** [comment]
Try to use password "0000" or None

**Reply by yozik04** [reply]
It might be that serial is encrypted. Look up an issue on git hub. People tell that it is possible to unlock serial port with a small payment.

**Comment by dsge80** [comment]
Hi. I've try it with 0000, none, aaaa ...
For sure communication is encrypted for this..
SP7000+ works correctly? I believe they are using the same encryption.
Thank you.

**Comment by dsge80** [comment]
Regarding the second part it's says also in their manual :) :

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/fde5a7fd-b4a7-4f27-9ff1-41dd69c76e86)

I'll see how I can buy it ...

Thank you

**Comment by dsge80** [comment]
I know I saw somewhere the discussion but I don't find it and in "Blueye" app I don't see any option to buy it.
Do someone recall the discussion or if possible to buy unlock key as an enduser?
Thanks

**Comment by lpaolini** [comment]
Unfortunately it's not possible for an end user to purchase an unlock code from Paradox.
Only installers (or people having an installer account) can do that... too bad.

**Reply by dsge80** [reply]
Indeed is not a good thing as they doesn't provide and easy option to unlock the serial for 3rd party.
Thank you.

**Comment by dsge80** [comment]
So I've continued to dig and I also updated the firmware to 1.4.1. It seems now is connecting partly.

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/d43c97f1-e063-4d09-a07f-60793e260c8f)
![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/53ff2fcb-3eaf-4c0f-b65f-c767c1c690af)
![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/4e85756c-b3a0-49a0-918c-898b897cc58d)

This is in MQTT Explorer:

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/0f649ca2-ed06-4ecd-aba8-5b0b5537b41f)

**Reply by yozik04** [reply]
Nope it does not connect. Same Auth error.

**Reply by dsge80** [reply]
Thank you for reply. 
SP7000+ it is using a different encryption? Because for last version of PAI is mentioned as sit should work with EVOHD+ and 7000+. 
I think I can still can change the board, but I'm afraid about different firmware versions which might not work.

**Reply by yozik04** [reply]
I think for SP7000+ you also need to buy a serial unlock.

**Reply by dsge80** [reply]
Is not I don't want to buy, unfortunately I can't as I don't have an professional installer for this. I bought the alarm to integrate it in HA for other smart projects and to cover the cables on wall , not for main purpose :D
So best recommendation for PAI , in case I don't succeed with the code,  with at least 8 zones would be then EVO192, right? It does depending on firmware? Can be rolled back to older version in case has some unrecognizable encryption?

**Reply by yozik04** [reply]
For EVO192, If it has firmware 7.70+ (with serial encryption) then you cannot downgrade. =)
But I do not understand why for some people it works out of the box and for some it does not. I do not see a pattern here.
Previously I was thinking that if you have IP150 then PAI always works. But with these PLUS board versions I am not sure now.

**Reply by dsge80** [reply]
So is need to find also for EVO192 fw to be under 7.70.
I also thought as if I'll have IP150 I would be "covered" . I've try it almost everything, including IP_CONNECTION_BARE ....
I'm curious about settings used by others with "+" versions if they can share. Also if someone with access to Paradox can provide/help with  the unlock code, I can pay "upfront" to their PayPal or Revolut card.
Thank you.

**Comment by dsge80** [comment]
Hello. I've unlocked serial port. But still I'm unable to connect apparently:

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/70b2cd47-22df-4b32-b3c2-87c691bed74d)

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/344e06d7-b177-4288-863e-3dc5426d5a30)


So possible I do something wrong.

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/b56827c9-f43f-464e-b115-cf7727093528)

- I've entered mainly 3 passwords, twice paradox, default for IP150. 
- Tried with actual PC pass and also with 0000

What else I can try? 

Thank you.

**Reply by yozik04** [reply]
You do everything right. Password was accepted. But for some reason panel responds with wrong status.
b'100600000016'

**Which decodes to:**
```
fields = Container: 
        data = b'\x10\x06\x00\x00\x00' (total 5)
        value = Container: 
            po = Container: 
                command = 1
                status = Container: 
                    reserved = False
                    alarm_reporting_pending = False
                    Winload_connected = False
                    NeWare_connected = False
            length = 6
            result = Container: 
                neware_answer = False
            callback = 0
        offset1 = 0
        offset2 = 5
        length = 5
    checksum = b'\x16' (total 1)
```

Can you try to connect with BabyWare. Does it connect?
If yes. Will you be able to dump traffic with wireshark and send me privately to email?

**Reply by yozik04** [reply]
How to dump is here: https://github.com/ParadoxAlarmInterface/pai/wiki/FAQ#capture-babyware---ip150-communication

**Reply by dsge80** [reply]
Hello. Yes is working without problems on Babyware. 
I've opened an issue few days ago. Decrypted dumps are there. (https://github.com/ParadoxAlarmInterface/pai/issues/410) as panel was reset it to factory default.
I'll send them also by email.

**Reply by yozik04** [reply]
Hah. YAML format has changed... Damn... need to tune decoder.

**Reply by dsge80** [reply]
I  was thinking is something wrong with the decoded file, but as this is something new for me, thought maybe I'm wrong... 😄 
In case you need other YAML files, please let me know.
Thank you for looking on this.

**Reply by yozik04** [reply, marked as answer]
Let's experiment a bit. Development build will finish in some minutes. I have removed strict checking if Winload_connected flag is True. Let's try and see if it will be happy with that and what new problems it will bring.

**Reply by dsge80** [reply]
It seems it WORKS 😄 

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/2c15312a-e36e-474a-9bfd-2d7c1a6b9a78)

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/98a459cc-babb-4660-978b-6e48b2762f22)

![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/ffa23c21-e893-4855-85f3-0127515e2371)
![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/c0daa05f-64d2-462e-b129-88d669508267)
![image](https://github.com/ParadoxAlarmInterface/pai/assets/55022237/81f6990e-9aba-4f51-a5ae-73e94a791509)

Battery reading has same behavior as for SP7000+ as far I saw.  But is ok from my side.

You are the best. :)

Now I need to integrate it and see the behavior, but is promising.

Thanks .

**Reply by dsge80** [reply]
I've done alarm installation and integration in HA today. Is working fine and changing of status is fast once serial communication is activated in Babyware.
I've added EVOHD+ as supported in Wiki.
Once again thank you to developers.

**Reply by yozik04** [reply]
Great to hear! Thanks for reporting back.

**Comment by mremedi2023** [comment]
My ip150 was updated to version 2024-04-25 - V6.10.010 and I couldn't connect to the evo anymore, can someone help me? thank you
